samples/excel/85-preview-apis/direct-precedents.yaml

```diff
@@ -0,0 +1,154 @@
+order: 6
+id: excel-direct-precedents
+name: Direct precedents
+description: 'This sample shows how to highlight precedent cells which directly affect the value of the currently selected cell, using the direct precedents API.'
+host: EXCEL
+api_set:
+    ExcelAPI: '1.12'
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
+        $("#select-first-cell").click(() => tryCatch(selectFirstCell));
+        $("#select-second-cell").click(() => tryCatch(selectSecondCell));
+        $("#get-direct-precedents").click(() => tryCatch(getDirectPrecedents));
+        $("#clear-highlighting").click(() => tryCatch(setup));
+
+        /* Create two sample tables with functions that span the tables. */
+        async function setup() {
+          await Excel.run(async (context) => {
+            context.workbook.worksheets.getItemOrNullObject("Sample2019Data").delete();
+            context.workbook.worksheets.getItemOrNullObject("Sample2020Data").delete();
+
+            // Set up the first sample table.
+            const sheet1 = context.workbook.worksheets.add("Sample2019Data");
+            const data1 = [
+              ["Product", "Qty", "Unit Price", "Total Price 2019"],
+              ["Almonds", 2, 7.5, "=C3 * D3"],
+              ["Coffee", 1, 34.5, "=C4 * D4"],
+              ["Chocolate", 5, 9.56, "=C5 * D5"]
+            ];
+
+            const range1 = sheet1.getRange("B2:E5");
+            range1.values = data1;
+            range1.format.autofitColumns();
+
+            // Set up the second sample table.
+            const sheet2 = context.workbook.worksheets.add("Sample2020Data");
+            const data2 = [
+              ["Product", "Qty", "Unit Price", "Total Price 2020", "Average Annual Price"],
+              ["Almonds", 2, 8.0, "=C3 * D3", "=Sample2019Data!E3 + E3 / 2"],
+              ["Coffee", 1, 36.5, "=C4 * D4", "=Sample2019Data!E4 + E4 / 2"],
+              ["Chocolate", 5, 11.2, "=C5 * D5", "=Sample2019Data!E5 + E5 / 2"]
+            ];
+
+            // Style the tables.
+            const range2 = sheet2.getRange("B2:F5");
+            range2.values = data2;
+            range2.format.autofitColumns();
+
+            const header1 = range1.getRow(0);
+            header1.format.fill.color = "#4472C4";
+            header1.format.font.color = "white";
+
+            const header2 = range2.getRow(0);
+            header2.format.fill.color = "#4472C4";
+            header2.format.font.color = "white";
+
+            sheet1.activate();
+            sheet2.activate();
+
+            await context.sync();
+          });
+        }
+
+        /* Select a cell with direct precedents on one worksheet. */
+        async function selectFirstCell() {
+          await Excel.run(async (context) => {
+            // Ensure correct worksheet is active.
+            const sheet = context.workbook.worksheets.getItem("Sample2020Data");
+            sheet.activate();
+            sheet.load("Sample2020Data");
```

Comment:
This `load` statement is unnecessary.

Reply:
Ok, thanks! Can you explain why?

Reply:
Sure!

You don't need a load because you're not actually getting any data from the workbook. The `getRange` method is just creating a proxy object that reflects that range. It's a reference to a range, not the contents of that range. To get values from it, you'd need to call `myRange.load("values")`. The same logic applies to worksheets. You're getting a reference to a worksheet named "Sample2020Data", not the worksheet itself. The snippet doesn't check if that worksheet exists until the `context.sync` call. Effectively, your `selectFirstCell` method is a series of reference creations and write operations (`activate` and `select`). It never calls anything that'll read from the workbook (from the add-in's perspective).

Relatedly, `load` queues up a command to pull property values from the workbook to your add-in. The string argument is a property name. In this case, by passing something that's not a property name, it's actually not loading anything.